web_ui/frontend/components/layout/AuthenticatedContent.tsx

```diff
@@ -69,7 +69,7 @@ const AuthenticatedContent = ({
     if (data && checkAuthentication) {
       return checkAuthentication(data);
     } else {
-      return data?.authenticated !== undefined;
+      return !!data?.authenticated;
```

Comment:
Note this fixes another issue that I saw where the previous solution would consider you authenticated even if data.authenticated == false.